### ISSUE TITLE
The delta_fitness estimates have a negative skew

### ISSUE THREAD

**Issue by Haddox** [neherlab/SARS2-refined-fitness#4]
As I mentioned on slack, when plotting the updated fitness effects at the level of individual clusters, it looks to me like there might be a negative skew. The below plots specifically show data for synonymous mutations with predicted counts > 10, similar to Figure 5 from the manuscript. The distributions look pretty good, but most are shifted in a slightly negative direction, and some have larger negative tails. For `cluster == all`, the mean delta_fitness value is -0.23. In comparison, the mean is 0.00 from Figure 5.

![image](https://github.com/user-attachments/assets/ab6701d5-a658-4577-811c-832c5d514883)

@lucasesta, you responded: Most likely the negative skew we find here is due to the fact that I am using a different estimator for the fitness effects, namely the mean with respect to the posterior distribution rather than its mode.

I could see this as a potential explanation. It also seems to me like it would be good to do a reality check, where we make sure that we can reproduce Figure 5 from the paper using data from the pipeline. Another possibility is that the predicted counts are skewed for some reason.

Specifically, the reality check would be: for the set of synonymous mutations used to make Figure 5 (i.e., those from the curated dataframe with at least 10 expected counts or 10 predicted counts depending on the approach), plot the distribution of fitness estimates using three approaches:
* the naive approach (counts are not context aware and effects are estimated as a log ratio)
* the context-aware approach using the log ratio of counts
* the context-aware approach using the Bayesian counts

We'd expect the distributions of fitness effects from the first two approaches to look the same as the paper. If not, that would indicate some inconsistency with the predicted counts.

**Comment by lucasesta** [comment]
@Haddox, I proceeded performing the test you suggested. For completeness, I also added the two training sets we are currently using in this repository. Here follow the histograms summarizing the result of the analysis.

![syn_fit_curated](https://github.com/user-attachments/assets/3cd972d9-82b7-47df-b3e8-682f7624b9e8)

The three columns coincide with the different training sets. In particular, 'All' corresponds to the same dataset used to generate Fig. 5 from the paper. It looks like the context-aware approach using the log-ratio is the closest to have a zero mean, which makes sense, as it is the quantity the model is trained on. Indeed, without the filter on the predicted counts, such mean is exactly zero. On the other hand, it looks like the probabilistic approach is introducing a small negative skew, which is more pronounced when using the posterior mean rather than the maximum a posteriori.

I also checked that the general linear models inferred with the code in this repository and the one used for the paper are exactly the same on the aforementioned curated dataset (Fig. 5 of the paper). Furthermore, the sum of the difference of the absolute values of the predicted counts for the curated data obtained with the two codes is equal to 2.37e-7.

This would lead me to conclude that there is no inconsistency in how the predicted counts are computed in this repository and in the one of the paper.
